File: java/src/quick_sort/randomTestData.java
package quick_sort;

import java.util.Arrays;
import java.util.Random;

class randomTestData {
	private static final Random rand = new Random(0);

	int[] array, expectedArray;
	int randomIndex;

	randomTestData() {
		this(50);
	}

	randomTestData(int maxN) {
		int n = 1+rand.nextInt(maxN), max = rand.nextInt(maxN+1);
		array = new int[n];
		for (int i = 0; i < n; i++) {
			array[i] = rand.nextInt(1 + max);
		}
		expectedArray = array.clone();
		Arrays.sort(expectedArray);

		randomIndex = rand.nextInt(n);
	}
}

File: go/quick_sort/random_test_data.go
package quick_sort

import (
	"math/rand/v2"
	"sort"
)

type randomTestData struct {
	array, expectedArray []int
	randomIndex          int
}

var random = rand.New(rand.NewPCG(0, 0))

func newRandomTestData() *randomTestData {
	return newRandomTestDataN(50)
}

func newRandomTestDataN(maxN int) *randomTestData {
	var td randomTestData

	n, max := 1+random.IntN(maxN), random.IntN(maxN+1)
	td.array = make([]int, n)
	for i := range n {
		td.array[i] = random.IntN(1 + max)
	}
	td.expectedArray = append([]int{}, td.array...)
	sort.Ints(td.expectedArray)

	td.randomIndex = random.IntN(n)

	return &td
}
